modules/platforms/cpp/tests/client-test/main.cpp

```diff
@@ -40,9 +52,23 @@ void before_all() {
 }
 
 int main(int argc, char **argv) {
+    // Install signal handlers to clean up resources on early exit.
+    signal(SIGABRT, signal_handler);
+    signal(SIGINT, signal_handler);
+
+    ignite::IgniteRunner runner;
+
+    shutdown_handler = [&](int signal) {
+        std::cout << "Caught signal " << signal << " during tests" << std::endl;
+
+        runner.stop();
+
+        std::cout << "Resources cleanup OK" << std::endl;
+    };
```

Comment:
Why do we need this? As far as I see this will just kill the process without any cleanup. And I'm not sure why to intercept SIGABRT in any case.

Reply:
Actually, this may be a good idea, as in case tests aborted with `SIGSEGV` (by the way, should add it I guess), IgniteRunner will continue running as a separate process which is really bad. But I don't like how the code looks right now, it should be moved to separate function so the main function looks somehow like this:
```
set_process_abort_handler([&](int signal) {
    std::cout << "Caught signal " << signal << " during tests" << std::endl;

    runner.stop();
};
```

Also, we can use the same code for Linux and Windows, not only macOS.